### PR TITLE
Add skip-O0 plugin option

### PR DESCRIPTION
**Pull request by AndrasKovacs**
Background: I was trying to use inspection to check unboxing in [this file](https://github.com/AndrasKovacs/setoidtt/blob/fb9688475a777082ae7452aae0e4ce9c29351a1f/setoidtt/src/Eval.hs), which yields about 7000 lines of core with `-O2`, most of which is to be inspected. 

I've found that loading in ghci becomes unacceptably slow with inspection, goes from around 2 seconds to something which I don't have the patience to wait for. In contrast, compilation with `-O` and `-O2` is only about 10% slower with inspection.

A possible solution to my problem would be to put inspections in a different module than my definitions. However, I've found that `inspection-testing` can only follow referenced core definitions in the current module, so this doesn't work for me.

While this slowdown in interactive mode is probably worth to investigate, I think a very simple solution, which seems useful anyway, is to add a plugin option which disables inspection in the case of `-O0`. So that's what I did.

### CODE REVIEW

File: src/Test/Inspection/Plugin.hs [hunk 1]
```diff
@@ -26,6 +26,7 @@ import Test.Inspection.Core
 --
 -- * @-fplugin-opt=Test.Inspection.Plugin:keep-going@ to ignore a failing build
 -- * @-fplugin-opt=Test.Inspection.Plugin:keep-going-O0@ to ignore a failing build when optimisations are off
+-- * @-fplugin-opt=Test.Inspection.Plugin:skip-O0@ to skip inspections when optimisations are off
 -- * @-fplugin-opt=Test.Inspection.Plugin:quiet@ to be silent if all obligations are fulfilled
```
Comment (nomeata) [inline src/Test/Inspection/Plugin.hs:30]: Can you rephrase the documentation to make it clear that `quiet` is orthogonal to the other three, and of the the other  three, at most one should be given?

**Review by nomeata — COMMENTED** [review]
Looks good

File: src/Test/Inspection/Plugin.hs (hunk 1 above)
Comment (AndrasKovacs) [reply src/Test/Inspection/Plugin.hs]: I revised the docs.

**Comment by nomeata** [comment]
Do you need a release of this or are you find with this slipping into the next release (whenever that is)?

**Comment by AndrasKovacs** [comment]
Thanks for the quick response here! I'm fine with the next release.
